packages/core-services/src/types/ServiceClass.ts

```diff
@@ -61,8 +62,15 @@ export abstract class ServiceClass implements IServiceClass {
 		this.api = api;
 	}
 
-	getEvents(): Array<keyof EventSignatures> {
-		return this.events.eventNames() as unknown as Array<keyof EventSignatures>;
+	getEvents(): { eventName: keyof EventSignatures; listeners: { (...args: any[]): void }[] }[] {
+		return this.events.eventNames().map((eventName) => ({
+			eventName: eventName as unknown as keyof EventSignatures,
+			listeners: this.events.rawListeners(eventName) as { (...args: any[]): void }[],
+		}));
+	}
+
+	removeAllListeners(): void {
+		this.events.removeAllListeners();
```

Comment:
@sampaiodiego I've added this method to remove all the listeners at once when the instance is destroyed, do you believe this would be unnecessary maybe?